csrc/scheduler/mark_aliases.cpp

```diff
@@ -33,21 +33,22 @@ void markAliases(Fusion* fusion) {
 
   for (TensorView* out :
        ir_utils::filterByType<TensorView>(fusion->outputs())) {
-    if (fusion->getOutputAlias(out).type != AllocationType::New) {
+    // AllocationType::ReuseBuffer requires the output to be updated in place
+    // so it can't be computed as an alias.
+    if (fusion->getOutputAlias(out).type == AllocationType::ReuseBuffer) {
```

Comment:
Probably doesn't make a difference in practice. AllocationType::Evaluate is used for either aliases or ATen fallback. The latter is picked up by the ExprEvalScheduler, which doesn't call this function. I made the change only because it's easier to explain why we should skip ReuseBuffer.